### PR TITLE
TST: Add tests for Affine2D

### PR DESCRIPTION
**Pull request by QuLogic**
## PR summary

This is implicitly tested via all plots and layout. However, this makes it difficult to work on the transform stack directly (cf #22191), which these tests rectify.

I deliberately chose points/angles that were easy to work out in my head, so hopefully there are not any edge cases that were missed here. (Except for `skew`, for which I don't have an inherent idea of what that transform looks like, so I admit to just copy-pasting the existing output there.)

## PR checklist

- [n/a] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [x] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [n/a] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [n/a] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/coding_guide.html#new-features-and-api-changes)
- [n/a] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

File: lib/matplotlib/tests/test_transforms.py [hunk 1]
```diff
@@ -14,6 +14,340 @@
...
+        sy = Affine2D().scale(1, -2)
+        trans = Affine2D().scale(3, -2)
+        assert_array_equal((sx + sy).get_matrix(), trans.get_matrix())
+        assert_array_equal(trans.transform(self.single_point), [3, -2])
+        assert_array_equal(trans.transform(self.multiple_points),
+                           [[0, -4], [9, -6], [12, 0]])
+
+    def test_skew(self):
+        trans_rad = Affine2D().skew(np.pi / 12, np.pi / 12)
+        trans_deg = Affine2D().skew_deg(15, 15)
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_transforms.py:105]: Thinking a bit more, this should probably skew different angles for `x` and `y`.

**Review by ksunden — APPROVED** [review]

File: lib/matplotlib/tests/test_transforms.py (hunk 1 above)
Comment (QuLogic) [reply lib/matplotlib/tests/test_transforms.py]: Talked with @ksunden and decided to use a couple of back-calculated values that produce a reasonable set of results.
